### DISCUSSION TITLE
機能とかバグとか

### DISCUSSION THREAD

**Discussion by kphrx** [kPherox/ZenzaWatch#10] [Ideas]
要望とか提案とか。重要になりそうなやつとか原因わかったやつはissueにする

**Comment by kphrx** [comment]
大百科から

**Reply by kphrx** [reply]
> シークバーにサムネイルを表示機能がしてないの自分だけかな
[1218](https://dic.nicovideo.jp/b/a/zenzawatch/1201-#1218)

**Reply by kphrx** [reply]
> ｄアニメストアの動画を視聴する時だけzenzawatchを無効にすることって出来ますでしょうか？
ｄアニメストアの動画はバッファ外へのシークがやや遅いのでお願いします。 
[1381](https://dic.nicovideo.jp/b/a/zenzawatch/1381-#1381)

**Reply by kphrx** [reply]
> ・画面ロード時に、自動で「ここまで見た」で再生される機能
・ニコられの「コメント時間（99:99）から再生」対応
・「いいね」コメントのコピペ、またはURL時にクリック出来る対応(いいねに、URL埋め込んでる動画がたまにあるため)
[1466](https://dic.nicovideo.jp/b/a/zenzawatch/1441-#1466)


- 解決 #25
  > ・画面ロード時に、自動で「ここまで見た」で再生される機能
  ・ニコられの「コメント時間（99:99）から再生」対応

  「コメント時間（99:99）から再生」は元々対応するコードは存在してたが機能してなかったので修正
- 未解決
  > ・「いいね」コメントのコピペ、またはURL時にクリック出来る対応(いいねに、URL埋め込んでる動画がたまにあるため)

**Reply by kphrx** [reply]
> MylistPocketのタグや説明文の変更の反映が遅くて困ってる
なんとかならないかな
[1480](https://dic.nicovideo.jp/b/a/zenzawatch/1471-#1480)

**Reply by kphrx** [reply]
> chromeでZenzaが動かなくなった
[1570](https://dic.nicovideo.jp/b/a/zenzawatch/1561-#1570)

> ChromeもZENZAも最新版に更新済み
マイページや動画視聴ページでは普通に動くけど
検索結果やシリーズで開こうとすると大抵固まるな
たまに動くけど条件が謎
[1572](https://dic.nicovideo.jp/b/a/zenzawatch/1561-#1572)

> 何回か動画を開き直すと再生できるようになるな
再生失敗したタブを閉じるときに変な負荷がかかっているのか1秒ほどフリーズする。
\
2.6.3-fix-playlist.13 
[1592](https://dic.nicovideo.jp/b/a/zenzawatch/1591-#1592)

**Reply by kphrx** [reply]
> やっぱコメントが表示されない
zenzaはplaylist16版のキャッシュとクッキー含めて削除してから再インストールした
もちろんTampermonkeyもキャッシュ含めて再インストール済み
正常にリセット出来てるかはzenzaの設定が全部消えてるかで確認
この状態でzenza開くと右側のコメントリストは表示されてリストのコメントも流れてるんだけど、肝心の動画上には表示されない
コメントリストのコメントをダブルクリックするとそのコメントが流れてる時間にジャンプはできるが動画上にはコメントは出てない
もうこれ以上できることがない
助けてくれ
>
> chromeバージョン
バージョン: 100.0.4892.0（Official Build）dev （64 ビット）
[1688](https://dic.nicovideo.jp/b/a/zenzawatch/1681-#1688)

> あー、コメント表示されないのはどうやらDev版Chromeバージョン100の問題みたい
v98で試したらコメント出ました
UAをマイナーバージョンで固定してもだめだったので、もっと根本的な仕様っぽい
3/29にはマイナー版もv100になるので対応お願いします・・・
[1691](https://dic.nicovideo.jp/b/a/zenzawatch/1681-#1691)

**Reply by kphrx** [reply]
> 動画一覧がマイリストだと連続再生ボタンにZenボタン出てくるけど、
シリーズ一覧からとか概要欄のシリーズ一覧URLにはZenボタンが出てこない模様
[1466](https://dic.nicovideo.jp/b/a/zenzawatch/1771-#1775)

**Reply by kphrx** [reply]
> ニコニコが12/4に30分以上の動画投稿でも720p以上に対応してくれたけど、zenzawatchだと12/4以降に投稿された30分以上の動画でも相変わらず360p固定っぽい？ 
[2019](https://dic.nicovideo.jp/b/a/zenzawatch/2011-#2019)

**Reply by kphrx** [reply]
> dmc.nicoだわ
[2050](https://dic.nicovideo.jp/b/a/zenzawatch/2041-#2050)

> chromeだと↑のエラー出て動かない…… 火狐ならいけたけど
[2051](https://dic.nicovideo.jp/b/a/zenzawatch/2041-#2051)

> adblock切っても動かないな 原因がよくわからん
[2052](https://dic.nicovideo.jp/b/a/zenzawatch/2041-#2052)

Chrome Canary で試したけど再現しない。何？

**Reply by kphrx** [reply]
> コメント問題が落ち着いたっぽいところで、
>
> (いつからだっけ？)自分のコメントを削除できるようになってるんだけど、Zenzawatchでも対応してくれないかなぁ…
>
> うっかり誤字ったりしたときとか、公式プレイヤーで開き直して削除するという手間が…
[2088](https://dic.nicovideo.jp/b/a/zenzawatch/2071-#2088)

**Reply by kphrx** [reply]
> とりあえずマイリストの登録できなくなってるんだけどこれって俺だけか？
暇人いたら検証してほしい
ver 2.6.3-fix-playlist.39(Dev) 
[2159](https://dic.nicovideo.jp/b/a/zenzawatch/2131-#2159)

> すまん、なんか別動画で検証したら普通に動いたわ
登録できなかったやつは既にマイリスにあって、消す処理でエラー吐いてるっぽいってことはわかってソースコード見たけどそれ以上はあんまわからんかった 
[2160](https://dic.nicovideo.jp/b/a/zenzawatch/2131-#2160)

~users/me/watch-later の hasNext が真の時に掘れずに削除できてないとかか？~ 最後がその動画の時しかCONFLICT返さないしCONFLICT返すときは最新の記録だからこの推測は誤りだし過去を掘るのができないコードではなさそう

**Comment by kphrx** [comment]
プレイリストをソートしたり逆順にしたりの後でplay prev/nextのキーボードショートカット発火すると順番通りに流れない

**Comment by kphrx** [comment]
ニコられてるコメントの強調表示を量で重み付けとか無効にする機能。全部のコメがニコられて強調表示が機能してないケースとかに対応したい

公式の重み付け

ランク|ニコる
---|---
1|1以上
2|3以上
3|6以上
4|9以上

**Comment by kphrx** [comment]
`window.ZenzaLib`とか`window.MylistPocketLib`が`undefined`になってる

Firefox Nightly 109.0a1 (2022-12-04)
Violentmonkey BETA v2.13.3.8

**Reply by kphrx** [reply]
Firefox Beta 108.0b4, Violentmonkey v2.13.3 の環境では再現せず。Violentmonkey BETA v2.13.3.8をインストールして実行すると再現

Violentmonkeyのv2.13.4/v2.14.0 stable releaseで発現する可能性

**Comment by Auxilyrica** [comment]
マウスホイールでの音量調整にて、音量downはできるが音量upができない

https://github.com/kphrx/ZenzaWatch/blob/969c7aa61b30a52392e0ce7b6933cd822460c8c6/src/NicoVideoPlayer.js#L170-L176
上記場所のdeltaをみるとちょくちょくNaNが入ってきていて過剰にdownしているようでした

**Comment by kphrx** [comment]
コメント追加や削除に排他処理の欠落

https://x.com/kphrx/status/1822196032677048829

**Comment by kphrx** [comment]
HLS Support が外部サイトでの再生に対応できてない

https://x.com/kphrx/status/1761485408679149915
